Sources/GRPC/AsyncAwaitSupport/GRPCAsyncServerHandler.swift

```diff
@@ -323,7 +338,7 @@ internal final class AsyncServerHandler<
 
     switch self.interceptorStateMachine.interceptRequestMessage() {
     case .intercept:
-      self.interceptors?.receive(.message(request))
+      self.interceptors?.receive(.message(request)) // TODO: if we wrapped this in a withSpan, would that be correct?
```

Comment:
Not quite; a span wrapping `invokeUserHandler` would cover the user code. Instrumenting the gRPC bits 'surrounding' it would require starting a span where we extract the metadata and ending it in `sendInterceptedStatus` (I think, would need to check a bit further).